iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -0,0 +1,412 @@
+                            break;
+                        }
+
+                    default:
+                        _logger.LogWarning($"Property: Received an unrecognized property update from service:" +
+                                        $"\n[ {writableProperty.Key}: {writableProperty.Value} ].");
+                        break;
+                }
+            }
+        }
+
+        // The callback to handle target temperature property update requests for a component.
+        private async Task HandleTargetTemperatureUpdateRequestAsync(string componentName, double targetTemperature, long version, object userContext)
+        {
+            const string targetTemperatureProperty = "targetTemperature";
+            _logger.LogDebug($"Property: Received - component=\"{componentName}\", [ \"{targetTemperatureProperty}\": {targetTemperature}°C ].");
+
+            _temperature[componentName] = targetTemperature;
+            IWritablePropertyResponse writableResponse = _deviceClient
+                .PayloadConvention
+                .PayloadSerializer
+                .CreateWritablePropertyResponse(_temperature[componentName], CommonClientResponseCodes.OK, version, "Successfully updated target temperature.");
+
+            var reportedProperty = new ClientPropertyCollection();
+            reportedProperty.AddComponentProperty(componentName, targetTemperatureProperty, writableResponse);
+
+            ClientPropertiesUpdateResponse updateResponse = await _deviceClient.UpdateClientPropertiesAsync(reportedProperty);
+
+            _logger.LogDebug($"Property: Update - component=\"{componentName}\", {reportedProperty.GetSerializedString()} is {nameof(CommonClientResponseCodes.OK)} " +
+                $"with a version of {updateResponse.Version}.");
+        }
+
+        // The callback to handle command invocation requests.
+        private Task<CommandResponse> HandleCommandsAsync(CommandRequest commandRequest, object userContext)
+        {
+            // In this approach, we'll first switch through the component name returned and handle each component-level command.
+            // For the "default" case, we'll first check if the component name is null.
+            // If null, then this would be a top-level command request, so we'll switch through each top-level command.
+            // If not null, then this is a component-level command that has not been implemented.
+
+            // Switch through CommandRequest.ComponentName to handle all component-level commands.
+            switch (commandRequest.ComponentName)
+            {
+                case Thermostat1:
+                case Thermostat2:
+                    // For each component, switch through CommandRequest.CommandName to handle the specific component-level command.
+                    switch (commandRequest.CommandName)
+                    {
+                        case "getMaxMinReport":

```

Comment:
We use constants for the component name, do we want to do the same for the command names? The same goes for the `"targetTemperatureName` up on line 104

Reply:
I didn't do that since we'd received feedback from a doc writer that for small samples declaring the variable name closer to the code block where it is used makes more sense. It helps people avoid having to continuously scroll up and down.